BlenderMalt/MaltNodes/MaltNode.py

```diff
@@ -57,11 +69,26 @@ def update(self):
         if self.disable_updates:
             return
         self._disable_updates_wrapper(self.malt_update)
+    
+    def copy(self, node):
+        #Find the node from its node tree so we have access to the node id_data
+        for tree in bpy.data.node_groups:
```

Comment:
does not find the node when you copy using ctrl-c / ctrl-v (for whatever reason). Therefore `node.id_data` stays None and fails the rest of the copy function. This is especially crucial when copying from one node tree to another. I have not found a solution to this particular problem though. It might be necessary to take the node tree out of the equation when copying nodes which adds a whole other level of complexity for sure

Reply:
possible solutions:
1) if the only thing we need from the node tree is to get copies of the parameters maybe storing a copy of them at all times on the node itself (as a custom dict attribute or something) might be the way to do
2) going back to storing the attributes themselves on the node but this time as custom properties DIRECTLY on the node to keep the possibility to add drivers
3) basically the same as 2) but maybe instead of on the node they could be stored on the sockets instead

Reply:
I found an easier fix.
It won't work copies across blend files, but I guess it's not that bad.

Your proposed solutions would solve that too, but maintaining a fully synced copy of all node parameters would be a maintenance hell, and storing attributes directly in the node or the socket would require a full rewrite of the properties system.

Reply:
Ah yeah I just saw. Could this be used as an identifier to fetch the parameters in the first place? because of the issue where parameters get reset when renaming a node?

Reply:
The problem would be that the internal parameter name and the variable name in the generated source must match, so it would make the generated source code much less readable.

We would also need a different id generation method, but that's the lesser problem.

I think for renames we could simply store the current name in the msgbus callback and reset the callback every time there's a rename.
Then we can simply add a rename_property function in MaltPropertyGroup.
It's not super pretty, but doesn't sound too problematic.